src/main/java/sirius/biz/elastic/reindex/MoveIndexAliasJobFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.elastic.reindex;
+
+import sirius.biz.cluster.work.DistributedTaskExecutor;
+import sirius.biz.jobs.JobCategory;
+import sirius.biz.jobs.JobFactory;
+import sirius.biz.jobs.batch.BatchProcessJobFactory;
+import sirius.biz.jobs.batch.DefaultBatchProcessTaskExecutor;
+import sirius.biz.jobs.params.ElasticEntityDescriptorParameter;
+import sirius.biz.jobs.params.Parameter;
+import sirius.biz.jobs.params.StringParameter;
+import sirius.biz.process.ProcessContext;
+import sirius.biz.tenants.TenantUserManager;
+import sirius.db.es.Elastic;
+import sirius.db.mixing.EntityDescriptor;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+import sirius.web.security.Permission;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Implements a job which moves the alias which marks an active index to a desired destination index.
+ */
+@Register(classes = JobFactory.class)
+@Permission(TenantUserManager.PERMISSION_SYSTEM_TENANT)
+public class MoveIndexAliasJobFactory extends BatchProcessJobFactory {
+
+    @Part
+    private Elastic elastic;
+
+    private ElasticEntityDescriptorParameter entityDescriptorParameter =
+            (ElasticEntityDescriptorParameter) new ElasticEntityDescriptorParameter("ed",
+                                                                                    "$MoveIndexAliasJobFactory.descriptorParameter")
+                    .markRequired();
+    private StringParameter stringParameter =
+            new StringParameter("destination", "$MoveIndexAliasJobFactory.destinationParameter").markRequired();
+
+    @Override
+    protected String createProcessTitle(Map<String, String> context) {
+        return "MoveIndexAlias";
```

Comment:
something like Strings.apply("Movin Elasticsearch Index %s to %s, ....) - would be _very nice_